lib/membrane_http_adaptive_stream/storage.ex

```diff
@@ -197,14 +200,32 @@ defmodule Membrane.HTTPAdaptiveStream.Storage do
              &storage_impl.remove(track_id, &1, %{type: :header}, &2)
            ) do
       {result, impl_state} =
-        storage_impl.store(
-          track_id,
-          to_add_name,
-          buffer.payload,
-          metadata,
-          %{mode: :binary, type: to_add_type},
-          impl_state
-        )
+        Bunch.Enum.try_reduce(to_add, impl_state, fn %{
```

Comment:
nitpick: You could do the pattern matchc inside the function so that you don't have such huge indent after `fn`